### ISSUE TITLE
[BUG] Serviceworker shows wrong request.referrer in Playwright

### ISSUE THREAD

**Issue by jakobrosenberg** [microsoft/playwright#2415]
**Context:**
- Playwright Version: 1.0.2
- Operating System: Windows
- Node version: 12.14.1
- Browser: Chromium
- Extra: [any specific details about your environment]

**Describe the bug**
In Chrome, making a request from the path `http://localhost:5000/foo/bar`, correctly shows the referrer as `http://localhost:5000/foo/bar` in service workers.

In Playwrights Chromium, making the same request from the same path incorrectly shows the referrer as `http://localhost:5000/`. It should be `http://localhost:5000/foo/bar`.

**Comment by aslushnikov** [comment]
@jakobrosenberg it took us a while to get to this bug...

So since there's no snippet, I improvised a little to try to reproduce this.

Here's what I tried:
- installed playwright@1.0.2
- launched playwright chromium using the following script: 

```js
const {chromium} = require('playwright');

(async () => {
  const browser = await chromium.launch({
    headless: false,
    devtools: true,
  });
  const context = await browser.newContext();
  const page = await context.newPage();
  await page.goto('https://googlechrome.github.io/samples/service-worker/basic/');
})();
```
- Navigate devtools to the Network Panel
- Hit "load icon" button and check the request headers for the icon file in Service Worker

I did end up with a proper referer header there: 

![image](https://user-images.githubusercontent.com/746130/90859250-b1dfd280-e33c-11ea-8fcc-e75dc903c13d.png)

The same behavior happens with Playwright v1.3.0.

So what do I do wrong?
